docs/user/adding-and-reading-pdf-outlines.md

```diff
@@ -0,0 +1,216 @@
+# PDF Outlines
+
+PDF outlines—also known as bookmarks—provide a structured navigation panel in PDF readers. `pypdf` allows you to read, create, and modify both simple and deeply nested outlines.
+
+## Writing PDF Outlines
+
+To add outlines, use the `pypdf.PdfWriter.add_outline_item` method. This method returns a reference to the created outline, which you can use as a parent to create nested (hierarchical) bookmarks.
+
+### Adding a Simple Outline
+
+The following example shows how to add a single top-level bookmark. It uses the `clone_from` parameter to load an existing PDF and copy all its pages and metadata efficiently. We then add an outline item pointing to the first page (index `0`) and save the result.
+
+
+```{testsetup}
+pypdf_test_setup("user/pdf-outlines", {
+    "input.pdf": "../resources/input.pdf",
+})
+```
+
+```{testcode}
+from pypdf import PdfWriter
+
+# Open the input PDF and clone its content
+writer = PdfWriter(clone_from="input.pdf")
+
+# Add a top-level bookmark
+writer.add_outline_item(
+    title="Introduction",
+    page_number=0
+)
+
+# Save the updated PDF
+with open("input.pdf", "wb") as f:
+    writer.write(f)
+```
+
+![PDF outline output](simple-outline.png)
+
+
+### Adding Nested Outlines
+
+You can build hierarchies (like Chapter → Section) by passing the parent outline item to the `parent` parameter of a new item.
+
+In the example below, we create a root item "Introduction" and nest two sections under it.
+
+```{testcode}
+from pypdf import PdfWriter
+
+writer = PdfWriter(clone_from="input.pdf")
+
+# Add parent (Chapter)
+introduction = writer.add_outline_item(
+    title="Chapter 1",
+    page_number=1
+)
+
+# Add children (sections) nested under the introduction
+writer.add_outline_item(
+    title="Section 1.1",
+    page_number=1,
+    parent=introduction
+)
+
+writer.add_outline_item(
+    title="Section 1.2",
+    page_number=1,
+    parent=introduction
+)
+
+with open("input.pdf", "wb") as f:
+    writer.write(f)
+```
+
+![PDF outline output](nested-outline.png)
+
+
+### Advanced Styling and View Modes (Fit Options)
+
+You can customize the appearance and behavior of bookmarks using optional parameters. The `color` argument (a tuple of floats `0.0-1.0`) controls the text color, while `bold` and `italic` control the font style.
```

Comment:
Describing the parameters itself should be part of the API docs, not the user guide. You should restrict descriptions here to mention that these aspects can be changed, but how to define colors is out of scope here IMHO.